servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ProxyConnectConnectionFactoryFilter.java

```diff
@@ -74,51 +68,59 @@ private ProxyFilter(final ConnectionFactory<ResolvedAddress, C> delegate) {
+                        connectAddress, response.status()));
+            }
 
-                    DeferSslHandler deferSslHandler = channel.pipeline().get(DeferSslHandler.class);
-                    if (deferSslHandler == null) {
-                        return response.payloadBodyAndTrailers().ignoreElements().concat(failed(
-                                new IllegalStateException("Failed to find a handler of type " +
-                                        DeferSslHandler.class + " in channel pipeline.")));
+            final Channel channel = ((NettyConnectionContext) connection.connectionContext()).nettyChannel();
+            final SingleSource.Processor<C, C> processor = newSingleProcessor();
+            channel.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                @Override
+                public void userEventTriggered(final ChannelHandlerContext ctx, final Object evt) {
+                    if (evt instanceof SslHandshakeCompletionEvent) {
+                        SslHandshakeCompletionEvent event = (SslHandshakeCompletionEvent) evt;
+                        if (event.isSuccess()) {
+                            processor.onSuccess(connection);
+                        } else {
+                            processor.onError(event.cause());
+                        }
                     }
+                    ctx.fireUserEventTriggered(evt);
+                }
+            });
 
-                    deferSslHandler.ready();
+            final DeferSslHandler deferSslHandler = channel.pipeline().get(DeferSslHandler.class);
+            if (deferSslHandler == null) {
+                return failed(new IllegalStateException("Failed to find a handler of type " +
+                        DeferSslHandler.class + " in channel pipeline."));
+            }
+            deferSslHandler.ready();
 
-                    // There is no need to apply offloading explicitly (despite completing `processor` on the
-                    // EventLoop) because `payloadBody()` will be offloaded according to the strategy for the
-                    // request.
-                    return response.payloadBodyAndTrailers().ignoreElements().concat(fromSource(processor));
-                } else {
-                    return response.payloadBodyAndTrailers().ignoreElements().concat(
-                            failed(new ProxyResponseException("Bad response from proxy CONNECT " + connectAddress,
-                                    response.status())));
-                }
-            }));
+            // There is no need to apply offloading explicitly (despite completing `processor` on the
+            // EventLoop) because `payloadBody()` will be offloaded according to the strategy for the
+            // request.
+            return response.payloadBodyAndTrailers().ignoreElements().concat(fromSource(processor));
+        } catch (Throwable t) {
```

Comment:
This `try-catch` isn't required as it is called from the `flatMap` which has a `recoverWith`, so if anything does throw from here, `flatMap` will correctly terminate the control flow with the same error which will then be intercepted by `recoverWith()` closing the connection.